Bxmsdoc 5232 master (#90)
mramendi/kie-docs: doc-content/enterprise-only/openshift/secret-adminuser-create-proc.adoc

[id='secret-adminuser-create-proc']
:INCLUDECENTRAL!:
ifeval::["{context}"=="openshift-authoring"]
:INCLUDECENTRAL: {CENTRAL}
endif::[]
ifeval::["{context}"=="openshift-authoring-managed"]
:INCLUDECENTRAL: {CENTRAL}
endif::[]
ifeval::["{context}"=="openshift-freeform"]
:INCLUDECENTRAL: {CENTRAL} Monitoring
endif::[]
ifeval::["{context}"=="openshift-managed"]
:INCLUDECENTRAL: {CENTRAL} Monitoring
endif::[]
//immutable handled separately, so do not define INCLUDECENTRAL for it

= Creating the secret for the administrative user

> You must create a secret that contains the user name and password for a {PRODUCT} administrative user account. This secret is required for deploying {PRODUCT} using any template except the trial template. 

Reviewer: ```suggestion
You must create a generic secret that contains the user name and password for a {PRODUCT} administrative user account. This secret is required for deploying {PRODUCT} using any template except the trial template.
```
Maybe it should be mentioned that the secret keys must be KIE_ADMIN_USER and KIE_ADMIN_PWD

If you are using multiple templates to deploy components of {PRODUCT}, use the same secret for all these deployments. The components utilize this user account to communicate with each other.

ifdef::INCLUDECENTRAL[]

> You can also use this user account to log in to {INCLUDECENTRAL}. If you do not use RH-SSO or LDAP authentication, use this account to create other user accounts.

Reviewer: I don't understand this sentence.
If you do not use RH-SSO or LDAP, the user won't be able to create any more user accounts.
If you do use RH-SSO or LDAP, these credentials will be used by KIE Server and Business/Decision Central to communicate with each other and must exist in RH-SSO or LDAP with the same exact values as it is mentioned in the next sections.

endif::INCLUDECENTRAL[]
ifdef::PAM[]
ifeval::["{context}"=="openshift-immutable"]

> If you deploy the immutable monitoring template, you can also use this user account to log in to {CENTRAL} Monitoring. If you do not use RH-SSO or LDAP authentication, use this account to create other user accounts.

Reviewer: Same comment as before

endif::[]
endif::PAM[]

[IMPORTANT]
====
If you use RH-SSO or LDAP authentication, the same user with the same password must be configured in your authentication system with the `kie-server,rest-all,admin` roles for {PRODUCT}.
====

.Procedure

// the name is rhpam-credentials even for RHDM - this is the example value used in templates

> Use the `oc` command to generate a secret named `kie-admin-user-secret` from the user name and password:

Reviewer: ```suggestion
Use the `oc` command to generate a generic secret named `rhpam-credentials` from the user name and password:
```
Maybe mention the literal keys must be KIE_ADMIN_USER and KIE_ADMIN_PWD?

= Creating the secret for the administrative user

You must create a generic secret that contains the user name and password for a {PRODUCT} administrative user account. This secret is required for deploying {PRODUCT} using any template except the trial template. 

> The secret must contain the user name and password as literals. key name for the user name is `KIE_ADMIN_USER`. The key name for the password is `KIE_ADMIN_PWD`.

Reviewer: ```suggestion
The secret must contain the user name and password as literals. The key name for the user name is `KIE_ADMIN_USER`. The key name for the password is `KIE_ADMIN_PWD`.
```
